src/java.base/share/classes/sun/util/locale/LanguageTag.java

```diff
@@ -414,6 +407,50 @@ private boolean parsePrivateuse(StringTokenIterator itr, ParseStatus sts) {
         return found;
     }
 
+    public static String caseFoldTag(String tag) {
+        LanguageTag parsedTag = parse(tag, null);
```

Comment:
I think we don't need to create a field `parseSts` but provide a local one here and use it in `parse(tag, sts), as that parse status is only referred to within this method.